differ/differ.go

```diff
@@ -155,21 +167,122 @@ func (entry oneEntry) Diff(other oneEntry) (int, bool) {
 	} else if entry.OpCode != other.OpCode {
 		return 0, false
 	} else if entry.OpCode == gomemcached.UPR_MUTATION {
-		if entry.RevId != other.RevId {
+		entryRevId := entry.GetRevId()
+		otherRevId := other.GetRevId()
+		if entryRevId != otherRevId && entryRevId != 0 && otherRevId != 0 { //Darshan : RevId comparison should be revisited
 			return 0, false
-		} else if entry.Cas != other.Cas {
+		} else if entry.GetVersion() != other.GetVersion() {
 			return 0, false
 		} else if entry.Flags != other.Flags {
 			return 0, false
 		} else if !shaCompare(entry.BodyHash, other.BodyHash) {
 			return 0, false
 		} else if entry.Datatype != other.Datatype {
 			return 0, false
+		} else if !compareHlv(entry.Hlv, other.Hlv, entry.Cas, other.Cas, entry.ImportCas, other.ImportCas, entry.bucketUUID, other.bucketUUID) {
+			return 0, false
 		}
 	}
 	return 0, true
 }
 
+func compareHlv(hlv1, hlv2 *hlv.HLV, cas1, cas2, importCas1, importCas2 uint64, bucketUUID1, bucketUUID2 hlv.DocumentSourceId) bool {
```

Comment:
Just looking at the code, I can see that it could be problematic if HLV ever has changes.

This function seems to me would be better implemented in the XDCR "library"
For the `type CRMetadata struct` in `crMetadata.go`, add a new `Diff(other CRMetadata)` method?

This function takes in:
- HLVs
- CAS's
- imporCAS's

Unsure about "bucketUUIDs"

But can you think about whether or not it makes sense to do it here or in the library?

Reply:
Yeah,I'll think about it

Reply:
@darshan-na  - just for clarity. What is your decision?

Reply:
I think we can move it to the library. That way it is neater.
So I was thinking if we can add another field in the crMetadata struct for bucketUUID.